c7n/schema.py

```diff
@@ -318,14 +322,147 @@ def process_resource(type_name, resource_type, resource_defs):
     return {'$ref': '#/definitions/resources/%s/policy' % type_name}
 
 
-if __name__ == '__main__':
-    from c7n.resources import load_resources
+def resource_vocabulary():
+    vocabulary = {}
+    for type_name, resource_type in resources.items():
+        docs = {'actions': {}, 'filters': {}}
+
+        actions = []
+        for action_name, cls in resource_type.action_registry.items():
+            actions.append(action_name)
+            docs['actions'][action_name] = inspect.getdoc(cls)
+
+        filters = []
+        for filter_name, cls in resource_type.filter_registry.items():
+            filters.append(filter_name)
+            docs['filters'][filter_name] = inspect.getdoc(cls)
```

Comment:
if there are no docs on the derived class we want docs from the base class by default.

```
>>> class bar(object): 
...    """i'm a lumberjack, short and stout, tip me over"""
... 
>>> class foo(bar):
...     pass
... 
>>> 
>>> import inspect
>>> inspect.getdoc(foo)
>>> 
```